### PR TITLE
fix: reduce apiHost steps when using genx in prompt mode - FUI-2207

### PR DESCRIPTION
**Pull request by MrBrunoWolff**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[FUI-2207](https://genesisglobal.atlassian.net/browse/FUI-2207)

🤔 &nbsp; **What does this PR do?**

- Attempt on fixing "blank apiHost" after setting it in genx steps (non-prompt mode)

🚀 &nbsp; **Where should the reviewer start?**

See the `api.js` changes related to the `apiHost` question + how the value is set/used [here](https://github.com/genesiscommunitysuccess/blank-app-seed/pull/343/files#diff-f3b44cae9c48356e0cdcd428aecf336a498a1bc7eaca242d7a2598a5d75df2aeR14).

📑 &nbsp; **How should this be tested?**

First:

```
rm -rf blankappseedtest && npx -y @genesislcap/genx@latest init blankappseedtest --ref bw/api-host-fix
```

Then:

- set a custom API_HOST
- finish all the steps
- search for the value you used as API_HOST in the app, should be correctly set/ready to use

Repeat the above but without the `--ref` for this branch:

- Do the same, but on `main/prerelease` branch, "API_HOST" will remain the default... only works if setting in the actual genx command with `--apiHost` param.

✅ &nbsp; **Checklist**

- [x] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

**Comment by cistov** [comment]
What are the steps to reproduce the api host issue?

```
> npx -y @genesislcap/genx@latest init myapp
ℹ Downloading NPM seed @genesislcap/blank-app-seed (dist tag latest, subdir: N/A) to /Users/artur.cistov/code/myapp directory...                               15:57:20
✔ Seed downloaded in 4181ms                                                                                                                                    15:57:24

You can override the default Genesis server URL used during local developent.
? Set API Host Yes
? API Host (with WebSocket prefix and suffix if any) ws://foo
Optionally, you can enable the SSO function, which will redirect to identity provider before starting the application (given we only have one identity provider).
? Init SSO connection before loading application No
? Project Description

? Group Id global.genesis
? Application Version 1.0.0-SNAPSHOT
? Enable deploy plugin? No
? Generate empty CSV for entities? (config in JSON format) []
? Framework Web Components
? UI configuration in JSON format {}
? Pages config in JSON format [{"name":"home"}]
✔ Project successfully created in 150123ms.                                                                                                                    15:59:50

ℹ (experimental) Opening shell in  myapp...                                                                                                                     15:59:50
> cat client/package.json| grep foo
    "API_HOST": "ws://foo",
```

**Comment by MrBrunoWolff** [comment]
riiight.. 

```bash
? Set API Host Yes
? API Host (with WebSocket prefix and suffix if any) ws://foo
```

I guess this was just me being "dumb" when using the prompts BUT... still think that we should just have ONE step for API_HOST (change proposed here)

Why do we have 2 steps to "fill or not fill API_HOST" when all the other prompts are "one-time" (if you specify that means the YES + the custom value or use the default)

@cistov

**Comment by MrBrunoWolff** [comment]
@cistov do you think it's worth continuing with this one?

File: .genx/prompts/api.js [hunk 1]
```diff
@@ -7,19 +7,12 @@ const ssoIntro = () => console.log(TEXTS.INTRO_API_SSO);
 module.exports = async (inquirer, prevAns = {}) => {
   apiHostIntro();
   const { apiHost = prevAns.apiHost } = await inquirer.prompt([
-    {
-      name: 'setApiHost',
-      type: 'confirm',
-      message: TEXTS.MESSAGE_API_SET_HOST,
```
Comment (cistov) [inline .genx/prompts/api.js:13]: Can get delete that `TEXTS.MESSAGE_API_SET_HOST` now

File: .genx/prompts/api.js [hunk 2]
```diff
@@ -7,19 +7,12 @@ const ssoIntro = () => console.log(TEXTS.INTRO_API_SSO);
...
-      default: true,
-    },
     {
       name: 'apiHost',
       type: 'input',
       message: TEXTS.MESSAGE_API_HOST,
-      when: ({ setApiHost }) => setApiHost,
-      default: prevAns.apiHost || 'ws://localhost/gwf/',
+      when: !prevAns.apiHost,
+      default: 'ws://localhost/gwf/',
```
Comment (cistov) [inline .genx/prompts/api.js:15]: We can update the default to `ws://localhost:9064` while we're at it. That's what Create uses. It's more useful.

**Comment by cistov** [comment]
> @cistov do you think it's worth continuing with this one?

@MrBrunoWolff I've finally got a chance to test it. All looking good. Just some minor comments.

File: .genx/prompts/api.js (hunk 1 above)
Comment (MrBrunoWolff) [reply .genx/prompts/api.js:13]: AHH indeed! good one

File: .genx/prompts/api.js (hunk 2 above)
Comment (MrBrunoWolff) [reply .genx/prompts/api.js:15]: will do

Comment (MrBrunoWolff) [reply .genx/prompts/api.js]: addressed in https://github.com/genesiscommunitysuccess/blank-app-seed/pull/343/commits/d54a0373aa21f0883c7d80b7f74dc586ccbcec3b

File: .genx/prompts/api.js (hunk 1 above)
Comment (MrBrunoWolff) [reply .genx/prompts/api.js:13]: addressed in https://github.com/genesiscommunitysuccess/blank-app-seed/pull/343/commits/d54a0373aa21f0883c7d80b7f74dc586ccbcec3b

**Comment by MrBrunoWolff** [comment]
@cistov this should be good to go now

I'll add a PR on `docs` repo too and link to this

**Review by cistov — APPROVED** [review]

**Pull request merged**